app/controllers/user_preferences_controller.rb

```diff
@@ -0,0 +1,29 @@
+class UserPreferencesController < ApplicationController
+
+  before_action :authenticate_user!
+  before_action :check_acting_as
+
+  def index
+    UserPreference.create_appropriate_user_preferences(current_user)
```

Comment:
Maybe `create_missing_user_preferences` might be a better name? "Appropriate" doesn't have much semantic meaning to me here. Not a big deal, though.